src/styling.rs

```diff
@@ -0,0 +1,788 @@
+/// [StandardizedVariants.txt]: <https://www.unicode.org/Public/UNIDATA/StandardizedVariants.txt>
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, Default)]
+pub enum MathStyle {
+    /// Unstyled default. May be serif or sans-serif depending on the font.
+    #[default]
+    Plain,
+    /// Bold style. May be serif or sans-serif depending on the font.
+    ///
+    /// Supported characters: digits, latin, greek, digamma.
+    Bold,
+    /// Italic style. May be serif or sans-serif depending on the font.
+    ///
+    /// Supported characters: latin, greek, dotless, and the extra ħ (U+0127).
+    Italic,
+    /// Bold italic style. May be serif or sans-serif depending on the font.
+    ///
+    /// Supported characters: latin, greek.
+    BoldItalic,
+    /// Script style.
+    ///
+    /// Supported characters: latin.
+    Script,
+    /// Bold script style.
+    ///
+    /// Supported characters: latin.
+    BoldScript,
+    /// Fraktur style. Also known as black-letter style.
+    ///
+    /// Supported characters: latin.
+    Fraktur,
+    /// Bold fraktur style. Also known as bold black-letter style.
+    ///
+    /// Supported characters: latin.
+    BoldFraktur,
+    /// Sans-serif style.
+    ///
+    /// Supported characters: digits, latin.
+    Sansserif,
+    /// Bold sans-serif style.
+    ///
+    /// Supported characters: digits, latin, greek.
+    SansserifBold,
+    /// Italic sans-serif style.
+    ///
+    /// Supported characters: latin.
+    SansserifItalic,
+    /// Bold italic sans-serif style.
+    ///
+    /// Supported characters: latin, greek.
+    SansserifBoldItalic,
+    /// Monospace style.
+    ///
+    /// Supported characters: digits, latin.
+    Monospace,
+    /// Isolated style.
+    ///
+    /// Supported characters: arabic excluding ه (U+0647), arabic-dotless.
+    Isolated,
+    /// Initial style.
+    ///
+    /// Supported characters: arabic excluding ا (U+0627), د–ز
+    /// (U+062F..U+0632), ط (U+0637), ظ (U+0638), و (U+0648).
+    Initial,
+    /// Tailed style.
+    ///
+    /// Supported characters: arabic excluding ا (U+0627), ب (U+0628),
+    /// ت (U+062A), ث (U+062B),  د–ز (U+062F..U+0632), ط (U+0637), ظ (U+0638),
+    /// ف (U+0641), ك (U+0643), م (U+0645), ه (U+0647), و (U+0648), and
+    /// arabic-dotless excluding ٮ (U+066E), ڡ (U+06A1).
+    Tailed,
+    /// Stretched style.
+    ///
+    /// Supported characters: arabic excluding ا (U+0627), د–ز
+    /// (U+062F..U+0632), ل (U+0644), و (U+0648), and arabic-dotless excluding
+    /// ٯ (U+066F), ں (U+06BA).
+    Stretched,
+    /// Looped style.
+    ///
+    /// Supported characters: arabic excluding ك (U+0643).
+    Looped,
+    /// Double-struck style. Also known as open-face style or blackboard-bold
+    /// style.
+    ///
+    /// Supported characters: digits, latin, arabic excluding ا (U+0627),
+    /// ك (U+0643), ه (U+0647), and the extras ∑ (U+2211), Γ (U+0393), Π
+    /// (U+03A0), γ (U+03B3), π (U+03C0).
+    Doublestruck,
```

Comment:
Same as with `Sansserif` here.
Either
```suggestion
 DoubleStruck,
```
or I'd also propose just using
```suggestion
 Double,
```

Reply:
Simply "double" instead of "double struck" isn't nearly as common as "sans" for "sans serif", so I think we should keep "double struck" here.

Reply:
The name used is double-struck in Unicode, so I figure its best to stick to that.